src/components/QueryBuilder/hooks/useHumanReadableJsonLogic.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback, ReactNode, ReactElement } from 'react';
+import {
+    Config,
+    ImmutableTree,
+    Utils as QbUtils,
+    Fields,
+    JsonLogicTree,
+    JsonGroup,
+} from 'react-awesome-query-builder';
+import isEmpty from 'lodash/isEmpty';
+
+import { useTranslatedConfig } from './useTranslatedConfig';
+import { QueryBuilderListToReplace } from '../types';
+
+const queryValue: JsonGroup = { id: QbUtils.uuid(), type: 'group' };
+
+const getColor = (
+    part: string,
+    listToReplace: QueryBuilderListToReplace[],
+): string => {
+    const color: string =
+        listToReplace.find(toReplaceConfig =>
+            toReplaceConfig.items.some(item => item === part),
+        )?.color || 'inherit';
+    return color;
+};
+
+export const useHumanReadableJsonLogic = (
+    fields: Fields,
+    listToReplace?: QueryBuilderListToReplace[],
+): ((
+    // eslint-disable-next-line no-unused-vars
+    logic?: JsonLogicTree,
+) => ReactNode) => {
+    const translatedConfig = useTranslatedConfig();
+    const getHumanReadableJsonLogic = useCallback(
+        (logic?: JsonLogicTree) => {
+            let queryString: any = '';
+            if (!isEmpty(fields) && logic) {
```

Comment:
I would invert the conditional:
```
if(isEmpty(fields) || !logic) return ''
// rest of the code
```

That way you don't have to reassign `queryString`, and the main logic of your function is not in a `if` statement